src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -183,11 +183,12 @@ public TypoSpellChecker(Context context)
 
       // subscribe to spelling prefs changes (invalidateAll on changes)
       ValueChangeHandler<Boolean> prefChangedHandler = (event) -> context_.invalidateAllWords();
+      ValueChangeHandler<Boolean> realtimeChangedHandler = (event) -> loadDictionary();
       ValueChangeHandler<String> dictChangedHandler = (event) -> loadDictionary();
-      userPrefs_.realTimeSpellchecking().addValueChangeHandler(prefChangedHandler);
       userPrefs_.ignoreUppercaseWords().addValueChangeHandler(prefChangedHandler);
       userPrefs_.ignoreWordsWithNumbers().addValueChangeHandler(prefChangedHandler);
       userPrefs_.spellingDictionaryLanguage().addValueChangeHandler(dictChangedHandler);
+      userPrefs_.realTimeSpellchecking().addValueChangeHandler(realtimeChangedHandler);
```

Comment:
If the app had started up with real time turned off, turning it on without restarting the app wouldn't then call `loadDictionary()` to ensure that Typo.js was loaded.